### PR TITLE
Introducing Custom Metrics plugin

### PR DESCRIPTION
**Pull request by MalcolmSmithUK**

### CODE REVIEW

File: tests/unit/04-plugins-customMetric.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+							}
+						}]
+					}
+				});
+				BOOMR.plugins.CustomMetric.parseRules(original);
+
+				assert.deepEqual(original, expected);
+			});
+
+			it("should modify the beacon given a malformed callback and a valid callback", function() {
```
Comment (tabanliviu) [inline tests/unit/04-plugins-customMetric.js:89]: @MalcolmSmithUK COVERAGE - add unit tests for null, undefined and NaN

**Review by tabanliviu — COMMENTED** [review]
Increase Coverage
